docs/topics/whatsnew-eap.md

```diff
@@ -87,11 +84,153 @@ We would appreciate any feedback you may have!
+* Report any problems you face with the new K2 compiler
+  in [our issue tracker](https://kotl.in/issue).
 * [Enable the **Send usage statistics** option](https://www.jetbrains.com/help/idea/settings-usage-statistics.html) to
   allow JetBrains to collect anonymous data about K2 usage.
-* 
+
+## Improved Gradle dependency handling for CInteropProcess in Kotlin/Native
+
+In this release, we enhanced the handling of the `defFile` property to ensure better Gradle task dependency management in 
+Kotlin/Native projects. 
+
+Before this update, Gradle builds could fail if the `defFile` property was designated as an output 
+of another task that hadn't been executed yet. The workaround for this issue was to add a dependency on this task:
+
+```kotlin
+kotlin {
+    macosArm64("native") {
+        compilations.getByName("main") {
+            cinterops {
+                val cinterop by creating {
+                    defFileProperty.set(createDefFileTask.flatMap { it.defFile.asFile })
+                    project.tasks.named(interopProcessingTaskName).configure {
+                        dependsOn(createDefFileTask)
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+To fix this, there is a new `RegularFileProperty` called `definitionFile`.  Now, Gradle lazily verifies the presence of 
+the `definitionFile` property after the connected task has run later in the build process. This new approach eliminates 
+the need for additional dependencies.
+
+The `CInteropProcess` task and the `CInteropSettings` class use the `definitionFile` property instead of `defFile` and 
+`defFileProperty`:
+
+<tabs>
+
+<tab id="kotlin" title="Kotlin">
+
+```kotlin
+kotlin {
+    macosArm64("native") {
+        compilations.getByName("main") {
+            cinterops {
+                val cinterop by creating {
+                    definitionFile.set(project.file("def-file.def"))
+                }
+            }
+        }
+    }
+}
+```
+
+</tab>
+
+<tab id="groovy" title="Groovy">
+
+```groovy
+kotlin {
+    macosArm64("native") {
+        compilations.main {
+            cinterops {
+                cinterop {
+                    definitionFile.set(project.file("def-file.def"))
+                }
+            }
+        }
+    }
+}
+```
+
+</tab>
+
+</tabs>
+
+> `defFile` and `defFileProperty` parameters are now deprecated.
+>
+{type="warning"}
+
+## Visibility changes in Gradle
+
+> This change doesn't impact Groovy DSL users.
+> 
+{type="note"}
+
+In Kotlin %kotlinEapVersion%, we've modified the Kotlin Gradle Plugin for better control and safety in your build scripts.
+Previously, certain Kotlin DSL functions and properties intended for a specific DSL context would inadvertently leak to 
+other DSL contexts. This leakage could lead to the use of incorrect compiler options, settings being applied multiple times,
+and other misconfigurations:
+
+```kotlin
+kotlin {
+  // Couldn't access methods and properties defined in the 
```

Comment:
Comments are not indented properly

Reply:
Yes. I was trying to wrap the comments so that the user didn't need to scroll. I've put the indents back and wrapped a bit more.